datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -766,7 +769,7 @@ protected void updateConsumerAssignment(Collection<TopicPartition> partitions) {
   public void onPartitionsRevoked(Collection<TopicPartition> topicPartitions) {
```

Comment:
`KafkaMirrorMakerConnectorTask` is overriding this method. You will have to fix this logic there as well. Without that this change is incomplete.

Reply:
KafkaMirrorMakerConnectorTask is calling super.Same with onPartitionsRevoked.  So I am thinking change in AbstractKafkaBasedConnectorTask should take care of it. COuld you pls confirm? 

 @Override
  public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
    super.onPartitionsRevoked(partitions);
    _topicManager.onPartitionsRevoked(partitions);
  }
@override
  public void run() {
    if (_enablePartitionAssignment) {
      try {
        LOG.info("Trying to acquire the lock on datastreamTask: {}", _datastreamTask);
        _datastreamTask.acquire(LOCK_ACQUIRE_TIMEOUT);
      } catch (DatastreamRuntimeException ex) {
        LOG.error(String.format("Failed to acquire lock for datastreamTask %s", _datastreamTask), ex);
        _dynamicMetricsManager.createOrUpdateMeter(generateMetricsPrefix(_connectorName, CLASS_NAME), _datastreamName,
            TASK_LOCK_ACQUIRE_ERROR_RATE, 1);
        throw ex;
      }
    }
    super.run();
  }

Reply:
At the current code state, yes, it will take care of it. But, it also has capability to create bigger problems in future. By not handling in the override class and relying that the logic is present in super class, in the future if someone changed the override class and remove the super call,  it will expose this problem. Since the current logic is very much dependent on how the code is written and it can change in future, it is critical to handle the scenario for easier debugging in future.

There are two ways to look at this:
a. TopicManager.onPartitionsRevoked will still be called which can add additional processing time based on what the implementation of that interface is. If it is decided that the subsequent calls should not be made, then additional checks are required in main class as well.
b. If it is not critical and we want to let the future users know about it, it is better to add a comment in KafkaMirrorMakerConnectorTask to not remove super.onPartitionsRevoked call or refactor the code to take care of the new kafka client scenario. 

I am very sure this knowledge will get lost in the future if there are not enough hints in the code.

Reply:
_topicManager.OnPartitionsRevoked is pretty much a noOp today. May be we could revisit this  overridden code onPartitionsRevoked in KafkaMirrorMakerConnectorTask later if needed completly.
For now, we can add comments to explicitly state the changes in upgraded kafka even in the overridden class as mentioned in the base class for clarity.
I will add this into a separate PR following this. Thanks!